apps/inheritance_app/inheritance_priv.h

```diff
@@ -16,10 +16,30 @@
 #include <inheritance/core.pb.h>
 #include <stdint.h>
 
+#include "card_fetch_data.h"
 #include "inheritance_context.h"
+#include "ui_input_text.h"
 
 /*****************************************************************************
  * TYPEDEFS
  *****************************************************************************/
 
+typedef struct {
+  secure_data_t data[INHERITANCE_MESSAGES_MAX_COUNT];
+  uint8_t data_count;
```

Comment:
Why is that preferred when we know that data_count can't get that high?

Reply:
particularly here because`data_count` is being copied from `plain_data_count` from query which is `pb_size_t`(which is >uint8_t) and can cause overflow

Reply:
Since both `uint8_t` and `pb_size_t` are unsigned The compiler will simple discard the bits that are not used during assignment. It won't/can't cause overflow, you can verify that [here](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) under section `6.3.1.8`. Even if an overflow was possible, the solution should've been to fix the overflow instead of updating the size.